version/version.go

```diff
@@ -10,5 +14,21 @@ var (
 
 	// Revision is filled with the VCS (e.g. git) revision being used to build
 	// the program at linking time.
-	Revision = ""
+	Revision string
 )
+
+func init() {
+	bi, ok := debug.ReadBuildInfo()
+	if !ok {
+		panic("no build info")
+	}
+	for _, s := range bi.Settings {
+		if s.Key == "vcs.revision" {
+			Revision = s.Value
+			break
+		}
+	}
+	if Revision == "" {
+		panic("unable to retrieve vcs revision")
```

Comment:
I think this is causing unit tests to fail since the `init` always gets run when importing this package. What if we didn't panic here and instead made `Revision` private (i.e. `revision`) and then have a `Revision()` func that returns it or an error to the caller if it's unset?